### ISSUE TITLE
running allennlp test-install after installing with pip leads to many errors

### ISSUE THREAD

**Issue by nelson-liu** [allenai/allennlp#1228]
After installing AllenNLP with: `pip install git+git://github.com/allenai/allennlp.git` and running `allennlp test-install`, a bunch of errors come up. Raising an issue just to enumerate them all. A full gist of the test traceback is here: https://gist.github.com/nelson-liu/78d08ad0090ad8f95c34675942f5bb4d

- The biggest issue is that we lose the directory structure in the repo. For example, `allennlp` with pip would get installed in `/home/nfliu/miniconda3/envs/test/lib/python3.6/site-packages/allennlp`. this contains _only_ the module (so it's [this folder](https://github.com/allenai/allennlp/tree/master/allennlp)).  The `tutorials`, `scripts`, `READMEs`, etc. aren't actually installed anywhere, so all of the paths in the tests break (e.g., script tests that shell out to run `perl ./scripts/some_eval_script.pl`).

- On a separate related note it looks like the precompiled custom extensions in the module aren't copied over either, maybe because they're not `.py` files and they get ignored automatically? This makes the `test_custom_extensions` / `alternating_lstm` tests break, even though they should work. This also means that custom extensions in general aren't working in the `pip install`ed version of allennlp.

## Things that are actually broken in the pip-installed version

- the custom highway lstm (since the extensions aren't included). fixed by #1229 
- `evalb_bracketing_scorer`, since `scripts/EVALB` does not exist. fixed by #1233 
- any tests that rely on files in `./tutorials` and `./scripts`. wip solution in #1232 

The other test errors are just errors with the tests themselves, not with the allennlp package. Which is unfortunate, but it doesn't affect _core_ library functionality.

**Comment by nelson-liu** [comment]
looks like `MANIFEST.in` controls what things go into the source distribution (the tarball) whereas the `package_data` key of `setup.py` controls what goes into the binary distribution / is copied over to `site-packages`...will try adding that and seeing what happens.

**Comment by nelson-liu** [comment]
To update on this: after #1233  and #1229 , `test-install` on a pip installed allennlp works for all tests except for the notebook tests and a SRL test and a span-based F1 test that compare python output to perl output. I'm disregarding the notebook tests as a wontfix since ostensibly they'll be rewritten in markdown anyway.

The bigger issue is how to access the tests from pip. Currently, they're inadvertently being installed as a module, which is unreliable behavior and pretty bad packaging. 
to summarize from #1232: 
- One solution would be to move the tests within the `allennlp` module (I could fix paths to be windows-compatible while I'm at it).
- Another solution would be to download and cache the appropriate test suite when `allennlp test-install` is run, and run the test on that downloaded version of the source.

Thoughts?
